### PR TITLE
Icon badge pixbuf

### PR DESCRIPTION
**Pull request by rrushton**
Some hard coded values, not sure best place to set them.

### CODE REVIEW

File: solus_sc/updates_view.py [hunk 1]
```diff
@@ -276,6 +277,53 @@ def load_updates(self):
...
+        corner_radius = height / 10.0;
+        degrees = pi / 180.0;
+
+        cr.arc(x + width - radius, y + radius, radius, -90 * degrees, 0 * degrees);
+        cr.arc(x + width - radius, y + height - radius, radius, 0 * degrees, 90 * degrees);
+        cr.arc(x + radius, y + height - radius, radius, 90 * degrees, 180 * degrees);
+        cr.arc(x + radius, y + radius, radius, 180 * degrees, 270 * degrees);
+
+    def create_icon_with_number(self, icon_in, count):
+        surface = cairo.ImageSurface (cairo.FORMAT_ARGB32, icon_in.get_width(), icon_in.get_height())
```
Comment (ikeydoherty) [inline solus_sc/updates_view.py]: This hasn't been run via `test.sh` has it?

File: solus_sc/updates_view.py [hunk 2]
```diff
@@ -276,6 +277,53 @@ def load_updates(self):
...
+        context = cairo.Context(surface)
+
+        Gdk.cairo_set_source_pixbuf(context, icon_in, 0, 0)
+        context.paint()
+
+        context.select_font_face("Clear Sans", cairo.FONT_SLANT_NORMAL, cairo.FONT_WEIGHT_BOLD)
```
Comment (ikeydoherty) [inline solus_sc/updates_view.py]: Not sure hard-coding the font is the best approach?

File: solus_sc/updates_view.py [hunk 3]
```diff
@@ -276,6 +277,53 @@ def load_updates(self):
...
+        context.set_font_size(14)
+
+        xbearing, ybearing, width, height, xadvance, yadvance = context.text_extents("{}".format(count))
```
Comment (ikeydoherty) [inline solus_sc/updates_view.py]: This line is far too long

File: solus_sc/updates_view.py [hunk 4]
```diff
@@ -276,6 +277,53 @@ def load_updates(self):
...
+
+        context.fill()
+
+        context.move_to(surface.get_width() - text_width, 15)
+
+        context.set_source_rgba(255,255,255,1)
+        context.show_text("{}".format(count))
+
+        surface = context.get_target()
+        icon = Gdk.pixbuf_get_from_surface(surface, 0, 0, surface.get_width(), surface.get_height())
```
Comment (ikeydoherty) [inline solus_sc/updates_view.py]: Too long. `test.sh` catches these issues :)

File: solus_sc/updates_view.py [hunk 5]
```diff
@@ -276,6 +277,53 @@ def load_updates(self):
...
+
+        context.set_source_rgba(199,0,0,1)
+
+        self.draw_rounded(context, surface.get_width() - width, 0, width, 20, 10)
+
+        context.fill()
+
+        context.move_to(surface.get_width() - text_width, 15)
+
+        context.set_source_rgba(255,255,255,1)
```
Comment (ikeydoherty) [inline solus_sc/updates_view.py]: Missing spaces after commas

File: solus_sc/updates_view.py [hunk 6]
```diff
@@ -504,6 +560,20 @@ def init_view(self):
                                       p_print, dlSize, icon, True, pkgSize,
                                       sc_obj])
 
+        count_normal = count_normal - count_security;
+
+        if (count_normal > 0):
```
Comment (ikeydoherty) [inline solus_sc/updates_view.py]: Unnecessary parentheses.

File: solus_sc/updates_view.py (hunk 2 above)
Comment (rrushton) [inline solus_sc/updates_view.py]: Yeah I made that initial note about hard coded values and not sure where to set/store them. For the font I will just use the system font.

Comment (ikeydoherty) [inline solus_sc/updates_view.py]: I think we should definitely include this in the next version of SC, but I'm not sure we'll both get enough time to land it in the first revision. You happy for me to add this to a TODO and a new issue? :)

Comment (rrushton) [inline solus_sc/updates_view.py]: Yeah I am easy so whatever works best. Fixed the font to use the current system font and the code formatting errors. (ran test.sh)

**Comment by ikeydoherty** [comment]
Sorry, unable to merge at present as I'm restructuring to remove gross inefficiencies.

**Pull request closed without merging**